app/durableObjects/ChatRoom.server.ts

```diff
@@ -367,35 +379,44 @@ export class ChatRoom extends Server<Env> {
 	async cleanupOldConnections() {
 		const meetingId = await this.getMeetingId()
 		if (!meetingId) log({ eventName: 'meetingIdNotFoundInCleanup' })
-		const websockets = this.ctx.getWebSockets()
+		const now = Date.now()
+		const users = await this.getUsers()
+		let removedUsers = 0
 		const connections = [...this.getConnections()]
-		log({
-			eventName: 'cleaningUpConnections',
-			meetingId,
-			connectionsFound: connections.length,
-			websocketsFound: websockets.length,
-			websocketStatuses: websockets.map((w) => w.readyState),
-		})
-		const sessionsToCleanUp = await this.getUsers()
-
-		connections.forEach((connection) =>
-			sessionsToCleanUp.delete(`session-${connection.id}`)
-		)
 
-		sessionsToCleanUp.forEach((user) => {
-			log({ eventName: 'userTimedOut', connectionId: user.id, meetingId })
-		})
+		for (const [key, user] of users) {
+			const connectionId = key.replace('session-', '')
+			const heartbeat = await this.ctx.storage.get<number>(
+				`heartbeat-${connectionId}`
+			)
+			if (heartbeat === undefined || heartbeat + 30_000 < now) {
```

Comment:
Here you use a hard codec interval value of 30_000, but further up you define an alarm interval of 15_000. Is that intentional? If so, maybe we should pur this value a global const as well?

Reply:
Ah, yep. These should be the same. Will fix!